multibody/kinematic/kinematic_evaluator_set.cc

```diff
@@ -134,6 +135,106 @@ std::vector<int> KinematicEvaluatorSet<T>::FindUnion(
   return union_index;
 }
 
+template <typename T>
+VectorX<T> KinematicEvaluatorSet<T>::CalcMassMatrixTimesVDot(
+    const Context<T>& context, const VectorX<T>& lambda) const {
+  // M(q)vdot + C(q,v) = tau_g(q) + F_app + Bu + J(q)^T lambda
+  VectorX<T> C(plant_.num_velocities());
+  plant_.CalcBiasTerm(context, &C);
+
+  VectorX<T> Bu = plant_.MakeActuationMatrix()
+      * plant_.get_actuation_input_port().Eval(context);
+
+  VectorX<T> tau_g = plant_.CalcGravityGeneralizedForces(context);
+
+  drake::multibody::MultibodyForces<T> f_app(plant_);
+  plant_.CalcForceElementsContribution(context, &f_app);
+
+  VectorX<T> J_transpose_lambda =
+      EvalFullJacobian(context).transpose() * lambda;
+
+  return tau_g + f_app.generalized_forces()+ Bu + J_transpose_lambda - C;
+}
+
+template <typename T>
+VectorX<T> KinematicEvaluatorSet<T>::CalcTimeDerivatives(
+    const Context<T>& context, const VectorX<T>& lambda) const {
+  MatrixX<T> M(plant_.num_velocities(), plant_.num_velocities());
+  plant_.CalcMassMatrix(context, &M);
+
+  VectorX<T> right_hand_side = CalcMassMatrixTimesVDot(context, lambda);
+
+  VectorX<T> v_dot = M.llt().solve(right_hand_side);
+
+  VectorX<T> x_dot(plant_.num_positions() + plant_.num_velocities());
+  VectorX<T> q_dot(plant_.num_positions());
+
+  plant_.MapVelocityToQDot(context, plant_.GetVelocities(context), &q_dot);
+
+  x_dot << q_dot, v_dot;
+
+  return x_dot;
+}
+
+template <typename T>
+VectorX<T> KinematicEvaluatorSet<T>::CalcTimeDerivatives(
+    const Context<T>& context, double alpha) const {
+  VectorX<T> lambda;
+  return CalcTimeDerivatives(context, &lambda, alpha);
+}
+
+template <typename T>
+VectorX<T> KinematicEvaluatorSet<T>::CalcTimeDerivatives(
+    const Context<T>& context, VectorX<T>* lambda, double alpha) const {
+  // M(q) vdot + C(q,v) = tau_g(q) + Bu + J(q)^T lambda
+  // J vdot + Jdotv  + kp phi + kd phidot = 0
+  // Produces linear system of equations
+  // [[M -J^T]  [[vdot  ]  =  [[tau_g + Bu - C     ]
+  //  [J  0 ]]  [lambda]]     [ -Jdotv - kp phi - kd phidot]]
+
+  // Evaluate manipulator equation terms
+  MatrixX<T> M(plant_.num_velocities(), plant_.num_velocities());
+  plant_.CalcMassMatrix(context, &M);
+
+  VectorX<T> C(plant_.num_velocities());
+  plant_.CalcBiasTerm(context, &C);
+
+  VectorX<T> Bu = plant_.MakeActuationMatrix()
+      * plant_.get_actuation_input_port().Eval(context);
+
+  VectorX<T> tau_g = plant_.CalcGravityGeneralizedForces(context);
+
+  drake::multibody::MultibodyForces<T> f_app(plant_);
+  plant_.CalcForceElementsContribution(context, &f_app);
+
+  // Evaluate constraint terms, phi/phidot for active. Jacboians for full
+  VectorX<T> phi = EvalActive(context);
+  VectorX<T> phidot = EvalActiveTimeDerivative(context);
+  MatrixX<T> J = EvalFullJacobian(context);
+  VectorX<T> Jdotv = EvalFullJacobianDotTimesV(context);
```

Comment:
Shouldn't the dimensions of `Jdotv`, `phi`, `phidot` be the same according to
  // [[M -J^T]  [[vdot  ]  =  [[tau_g + Bu - C     ]
  //  [J  0 ]]  [lambda]]     [ -Jdotv - kp phi - kd phidot]]
?

Reply:
Good catch! Will use active for all.